This is the deliverable for the tinder project of speed dating, for the Jedha data science & engineering certification bloc 2

Author : Youenn PATAT

![Tinder](https://full-stack-assets.s3.eu-west-3.amazonaws.com/M03-EDA/Tinder-Symbole.png)


# 1) Discovering of the general data.

The goal of this project is to see what factors are important to consider a second date.
For that we will follow the different steps that are note at titles.

Fisrt we will discover the dataset and see what we can do with it.

In [2]:
#import libraries

import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
from plotly.subplots import make_subplots

The first preview of the dataset.

In [3]:
#read the dataset
df = pd.read_csv("Speed_Dating_Data.csv", encoding = "ISO-8859-1")
pd.set_option('display.max_columns', None)

df.head()

iid   id  gender  idg  condtn  wave  round  position  positin1  order  \
0    1  1.0       0    1       1     1     10         7       NaN      4   
1    1  1.0       0    1       1     1     10         7       NaN      3   
2    1  1.0       0    1       1     1     10         7       NaN     10   
3    1  1.0       0    1       1     1     10         7       NaN      5   
4    1  1.0       0    1       1     1     10         7       NaN      7   

   partner   pid  match  int_corr  samerace  age_o  race_o  pf_o_att  \
0        1  11.0      0      0.14         0   27.0     2.0      35.0   
1        2  12.0      0      0.54         0   22.0     2.0      60.0   
2        3  13.0      1      0.16         1   22.0     4.0      19.0   
3        4  14.0      1      0.61         0   23.0     2.0      30.0   
4        5  15.0      1      0.21         0   24.0     3.0      30.0   

   pf_o_sin  pf_o_int  pf_o_fun  pf_o_amb  pf_o_sha  dec_o  attr_o  sinc_o  \
0      20.0      20.0      20.0       0.0       5.0      0     6.0     8.0   
1       0.0       0.0      40.0       0.0       0.0      0     7.0     8.0   
2      18.0      19.0      18.0      14.0      12.0      1    10.0    10.0   
3       5.0      15.0      40.0       5.0       5.0      1     7.0     8.0   
4      10.0      20.0      10.0      10.0      20.0      1     8.0     7.0   

   intel_o  fun_o  amb_o  shar_o  like_o  prob_o  met_o   age field  field_cd  \
0      8.0    8.0    8.0     6.0     7.0     4.0    2.0  21.0   Law       1.0   
1     10.0    7.0    7.0     5.0     8.0     4.0    2.0  21.0   Law       1.0   
2     10.0   10.0   10.0    10.0    10.0    10.0    1.0  21.0   Law       1.0   
3      9.0    8.0    9.0     8.0     7.0     7.0    2.0  21.0   Law       1.0   
4      9.0    6.0    9.0     7.0     8.0     6.0    2.0  21.0   Law       1.0   

  undergra mn_sat tuition  race  imprace  imprelig     from zipcode  \
0      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
1      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
2      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
3      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   
4      NaN    NaN     NaN   4.0      2.0       4.0  Chicago  60,521   

      income  goal  date  go_out  career  career_c  sports  tvsports  \
0  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
1  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
2  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
3  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   
4  69,487.00   2.0   7.0     1.0  lawyer       NaN     9.0       2.0   

   exercise  dining  museums  art  hiking  gaming  clubbing  reading   tv  \
0       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
1       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
2       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
3       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   
4       8.0     9.0      1.0  1.0     5.0     1.0       5.0      6.0  9.0   

   theater  movies  concerts  music  shopping  yoga  exphappy  expnum  \
0      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
1      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
2      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
3      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   
4      1.0    10.0      10.0    9.0       8.0   1.0       3.0     2.0   

   attr1_1  sinc1_1  intel1_1  fun1_1  amb1_1  shar1_1  attr4_1  sinc4_1  \
0     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
1     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
2     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
3     15.0     20.0      20.0    15.0    15.0     15.0      NaN      NaN   
4     15.0     

The first preview of this dataset's statistics.

In [114]:
df.describe(include="all")

iid           id       gender          idg       condtn  \
count   8378.000000  8377.000000  8378.000000  8378.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean     283.675937     8.960248     0.500597    17.327166     1.828837   
std      158.583367     5.491329     0.500029    10.940735     0.376673   
min        1.000000     1.000000     0.000000     1.000000     1.000000   
25%      154.000000     4.000000     0.000000     8.000000     2.000000   
50%      281.000000     8.000000     1.000000    16.000000     2.000000   
75%      407.000000    13.000000     1.000000    26.000000     2.000000   
max      552.000000    22.000000     1.000000    44.000000     2.000000   

               wave        round     position     positin1        order  \
count   8378.000000  8378.000000  8378.000000  6532.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      11.350919    16.872046     9.042731     9.295775     8.927668   
std        5.995903     4.358458     5.514939     5.650199     5.477009   
min        1.000000     5.000000     1.000000     1.000000     1.000000   
25%        7.000000    14.000000     4.000000     4.000000     4.000000   
50%       11.000000    18.000000     8.000000     9.000000     8.000000   
75%       15.000000    20.000000    13.000000    14.000000    13.000000   
max       21.000000    22.000000    22.000000    22.000000    22.000000   

            partner          pid        match     int_corr     samerace  \
count   8378.000000  8368.000000  8378.000000  8220.000000  8378.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean       8.963595   283.863767     0.164717     0.196010     0.395799   
std        5.491068   158.584899     0.370947     0.303539     0.489051   
min        1.000000     1.000000     0.000000    -0.830000     0.000000   
25%        4.000000   154.000000     0.000000    -0.020000     0.000000   
50%        8.000000   281.000000     0.000000     0.210000     0.000000   
75%       13.000000   408.000000     0.000000     0.430000     1.000000   
max       22.000000   552.000000     1.000000     0.910000     1.000000   

              age_o       race_o     pf_o_att     pf_o_sin     pf_o_int  \
count   8274.000000  8305.000000  8289.000000  8289.000000  8289.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      26.364999     2.756653    22.495347    17.396867    20.270759   
std        3.563648     1.230689    12.569802     7.044003     6.782895   
min       18.000000     1.000000     0.000000     0.000000     0.000000   
25%       24.000000     2.000000    15.000000    15.000000    17.390000   
50%       26.000000     2.000000    20.000000    18.370000    20.000000   
75%       28.000000     4.000000    25.000000    20.000000    23.810000   
max       55.000000     6.000000   100.000000    60.000000    50.000000   

           pf_o_fun     pf_o_amb     pf_o_sha        dec_o       attr_o  \
count   8280.000000  8271.000000  8249.000000  8378.000000  8166.000000   
unique          NaN          NaN          NaN          NaN          NaN   
top             NaN          NaN          NaN          NaN          NaN   
freq            NaN          NaN          NaN          NaN          NaN   
mean      17.459714    10.685375    1

In [115]:
df.shape

(8378, 195)

There are 8378 lines (participants) for 195 columns (answers/informations...). 

# 2) 1st step, The least desirable attribute.

### a) Least desirable attribute for male.

Firstly, we will take the data for the score of the different attributes in the second phase of questions, 1 day after the dating experience (Time 2). It's an arbitrary choice, because I think it is more realistic to have the score of attribute just after the experiment (Time 2) than before (Time 1) or weeks after (Time 3).

The name for the scored categories at Time 2 are the following: `attr7_2` for "Attractive", `sinc7_2` for "Sincere", `intel7_2` for "Intelligent", `fun7_2` for "Fun", `amb7_2` for "Ambitious" and `shar7_2` for "Has shared interests/hobbies". These categories are for 1 day after the date, the score of each attribute is between 0-100 and the total score is 100. It is asked to think what was important to say yes or no for a second date.

Like there are a lot of answers, data, we will use the mean of each categories' score for the male to see what attribute is the least desirable attribute and the most desirable attribute than they choose.

For the `gender` 0 is female and 1 is male.
 

In [116]:
list_of_attribut = ["attr7_2", "sinc7_2", "intel7_2", "fun7_2", "amb7_2", "shar7_2"]

# Filer the data where we have a male, gender = 1
df_male = df[df['gender'] == 1]

# calculate the mean for the attributes
attribute_for_men = df_male[list_of_attribut].mean().reset_index(name="Mean")
attribute_for_men = attribute_for_men.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr7_2': 'Attractivity',
    'sinc7_2': 'Sincerity',
    'intel7_2': 'Intelligence',
    'fun7_2': 'Fun',
    'amb7_2': 'Ambitious',
    'shar7_2': 'Shared interests/hobbies'
}

# Replace the attributes names
attribute_for_men['Attributes'] = attribute_for_men['Attributes'].replace(attribute_mapping)
attribute_for_men


Attributes       Mean
0              Attractivity  37.314525
1                 Sincerity  11.684266
2              Intelligence  14.986416
3                       Fun  19.342738
4                 Ambitious   6.386484
5  Shared interests/hobbies  10.394931

In [117]:
#Plot for the attributes considering by men
fig = px.pie(attribute_for_men, names = "Attributes", values = "Mean", color_discrete_sequence=px.colors.qualitative.Pastel1)
fig.update_layout(
    title = "Attributes' scores by men",
    autosize=False,
    width=700,
    height=512
)
fig.show()

We can see here than the more important attribute for men is the **attractivity** and the least one is the **ambitious**.

### b) Least desirable attribute for women.

Now we will do the same thing for womens.

In [118]:
list_of_attribut = ["attr7_2", "sinc7_2", "intel7_2", "fun7_2", "amb7_2", "shar7_2"]

# Filer the data where we have a female, gender = 0
df_female = df[df['gender'] == 0]

# calculate the mean for the attributes
attribute_for_women = df_female[list_of_attribut].mean().reset_index(name="Mean")
attribute_for_women = attribute_for_women.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr7_2': 'Attractivity',
    'sinc7_2': 'Sincerity',
    'intel7_2': 'Intelligence',
    'fun7_2': 'Fun',
    'amb7_2': 'Ambitious',
    'shar7_2': 'Shared interests/hobbies'
}

# Replace the attributes names
attribute_for_women['Attributes'] = attribute_for_women['Attributes'].replace(attribute_mapping)
attribute_for_women


Attributes       Mean
0              Attractivity  28.630964
1                 Sincerity  15.263889
2              Intelligence  15.580331
3                       Fun  18.426485
4                 Ambitious   8.132937
5  Shared interests/hobbies  13.779942

In [119]:
#Plot for the attributes considering by women
fig = px.pie(attribute_for_women, names = "Attributes", values = "Mean", color_discrete_sequence=px.colors.qualitative.Pastel1)
fig.update_layout(
    title = "Attributes' scores by women",
    autosize=False,
    width=700,
    height=512
)
fig.show()

For the womens, the more important attribute for men is the **attractivity** and the least one is the **ambitious**, the same as the men in general, with little changes in proportions.

# 3) 2nd step, Importance of the attractiveness.

Now we will see the importance of attractiveness in potential mate selection, before the date experience, and after the date, the real life impact.

For that we will check the score of attractiveness in Time 1 (during inscription before date): `attr1_1` for Attractive.


And in Time 2 (after the first date): `attr7_2` for "Attractive".

In [306]:
attractivity_before_date = df["attr1_1"].mean()
print("The mean of average score for attractiveness for everyone, before the dates, is:")
display(attractivity_before_date)

attractivity_after_date = df["attr7_2"].mean()
print("The mean of average score for attractiveness for everyone, after the dates, is:")
display(attractivity_after_date)

data = {
    'moment': ['before date (potential)', 'after date (real impact)'],
    'mean': [attractivity_before_date, attractivity_after_date]
}

attractivity_score_mean = pd.DataFrame(data)

display(attractivity_score_mean)

The mean of average score for attractiveness for everyone, before the dates, is:


22.51463188335944

The mean of average score for attractiveness for everyone, after the dates, is:


32.819556451612904

moment       mean
0   before date (potential)  22.514632
1  after date (real impact)  32.819556

In [286]:
fig = px.bar(attractivity_score_mean, x = "moment", y = "mean", color_discrete_sequence=px.colors.qualitative.Pastel2)
fig.update_layout(
    title = "Attractiveness for people before the date and after the date",
    xaxis_title = "Moment from the date",
    yaxis_title = "Mean score of attractiveness",
    autosize=False,
    width=1000,
    height=512
)

fig.show()


We see that people think **attractiveness** is less important in a potential way, before the date than the real impact of it, after the date to choose if they continu with the other people.

Now we will see if there is a difference if we seperate men and women thoughts.

In [276]:
df_female = df[df['gender'] == 0]

# calculate the mean for the attributes
data = {"Mean" : [df_female["attr1_1"].mean(), df_female["attr7_2"].mean()],
                  "moment" : ["before date (potential)", "after date (real impact)"]}
                  
attractivity_moment_women = pd.DataFrame(data)

display(attractivity_moment_women)

Mean                    moment
0  18.055224   before date (potential)
1  28.630964  after date (real impact)

In [277]:
df_male = df[df['gender'] == 1]

# calculate the mean for the attributes
data = {"Mean" : [df_male["attr1_1"].mean(), df_male["attr7_2"].mean()],
                  "moment" : ["before date (potential)", "after date (real impact)"]}
                  
attractivity_moment_men = pd.DataFrame(data)

display(attractivity_moment_men)

Mean                    moment
0  26.921689   before date (potential)
1  37.314525  after date (real impact)

In [284]:

trace1 = go.Bar(
    x=attractivity_moment_women["moment"],
    y=attractivity_moment_women["Mean"],
    name = str("Female")
)
trace2 = go.Bar(
    x=attractivity_moment_men["moment"],
    y=attractivity_moment_men["Mean"],
    name = str("Male")
)

fig = make_subplots()
fig.add_trace(trace1)
fig.add_trace(trace2)

fig.update_layout(xaxis_title='Moment from the date', yaxis_title='Mean of attractiveness', legend_title='Gender',
                autosize=False,
                width=1000,
                height=512 )

fig.show()

In [285]:
attractivity_before_date = df.groupby('gender')["attr1_1"].mean().reset_index(name="mean attractiveness before date")
attractivity_before_date["moment"] = ["before date (potential)"]*2
display(attractivity_before_date)

attractivity_after_date = df.groupby('gender')["attr7_2"].mean().reset_index(name="mean attractiveness after date")
attractivity_after_date["moment"] = ["after date (real impact)"]*2
display(attractivity_after_date)


gender  mean attractiveness before date                   moment
0       0                        18.055224  before date (potential)
1       1                        26.921689  before date (potential)

gender  mean attractiveness after date                    moment
0       0                       28.630964  after date (real impact)
1       1                       37.314525  after date (real impact)

So we can see than trend is the same for the 2 genders, before dating, the potential mean score importance of attractiveness is lower than the real impact after dating. Even if the proportion is higher for male than female to find attarctiveness more important.

# 4) 3rd step, shared interests Vs shared racial background.

In this section, we will try to compare the importance of the shared interests and the shared racial backgroud. Which one is more important for people to say yes to a second date.

For the shared racial background, we will use: `imprace` corresponding to the question "How important is it to you (on a scale of 1-10) that a person you date be of the same racial/ethnic background?", and there is this question only once, in Time 1, before the dating.

So for the shared interests, we will take the information at the same time: `shar1_1` for "Has shared interests/hobbies".

To have a nice comparison betwwen the 2 mean values, between 0 - 100 for the interests shared and between 0 - 10 for the racila shared, we compute the percentage of the ricial shared by dividing this by 10 and multiply it by 100.


In [312]:
interest_shared = df["shar1_1"].mean()
print("The mean of shared interest score is:")
display(interest_shared)

shared_racial = df["imprace"].mean()
print("The mean of average shared racial background score is:")
display(shared_racial/10 * 100) #calculation to put the value between 0 and 100 to have a percentage, like interests shared

data = {
    'category': ['Interests shared', 'Racial background shared'],
    'mean': [interest_shared, shared_racial/10 *100]
}

importance_interests_vs_racial = pd.DataFrame(data)

display(importance_interests_vs_racial)

The mean of shared interest score is:


11.845110815066004

The mean of average shared racial background score is:


37.847933485962166

category       mean
0          Interests shared  11.845111
1  Racial background shared  37.847933

In [322]:
fig = px.bar(importance_interests_vs_racial, x = "category", y = "mean", color = 'category', color_discrete_sequence=px.colors.qualitative.Pastel)
fig.update_layout(
    title = "Comparison of importance : <b> Interests shared Vs Racial background shared </b>",
    xaxis_title = "Category for comparison",
    yaxis_title = "Mean score of importance (%)",
    autosize=False,
    width=1000,
    height=512,
    showlegend=False
)

fig.show()

It seems than the racial background shared is more important than the shared interests. But we can have some word on these results du to how the data were in the dataset and how questions were asked.

First, as said before the scaling for the score note is not the same for the 2 categories, it is between 0 - 10 for the shared racial background and 0 - 100 for the shared interests. That means that we have done a calculation to put the shared racial background on the same scaling, between 0 - 100.
* Moreover, it has maybe include some bias in the results, because when you answer 2/10 for one thing, it can appears pretty small, wheraes answering 20% seems nit that small.
* The interests shared were directly compared to 5 other attributes to compared, so the score % depends on that. If people thought that other attributes directly compared were more important, it decreases instantly the score of the shared interests.
* The shared racial backgroud was just one question asking alone, without any other direct comparison.

# 5) 4th step, own perceived value

In this section, we will try to see if people perceived themselves correctly in comparison of the other perception.

For the own perception (in a scale on 0 - 10): `attr3_1` for "Attractive", `sinc3_1` for "Sincere", `int3_1` for "Intelligent", `fun3_1` for "Fun", `amb3_1` for "Ambitious".

For the other evaluation (in a scale on 0 - 10): `attr_o` for "Attractive", `sinc_0` for "Sincere", `int_0` for "Intelligent", `fun_0` for "Fun", `amb_0` for "Ambitious".

For the own evaluation:

In [4]:
list_of_attribut = ["attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1"]

# calculate the mean for the attributes
own_perception_general = df[list_of_attribut].mean().reset_index(name="Mean")
own_perception_general = own_perception_general.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr3_1': 'Attractivity',
    'sinc3_1': 'Sincerity',
    'intel3_1': 'Intelligence',
    'fun3_1': 'Fun',
    'amb3_1': 'Ambitious'
}

# Replace the attributes names
own_perception_general['Attributes'] = own_perception_general['Attributes'].replace(attribute_mapping)
own_perception_general


Attributes      Mean
0  Attractivity  7.084733
1     Sincerity  8.294935
2  Intelligence  8.403965
3           Fun  7.704460
4     Ambitious  7.578388

Evaluation by patners:

In [ ]:
list_of_attribut = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o"]

# calculate the mean for the attributes
partner_perception_general = df[list_of_attribut].mean().reset_index(name="Mean")
partner_perception_general = partner_perception_general.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr_o': 'Attractivity',
    'sinc_o': 'Sincerity',
    'intel_o': 'Intelligence',
    'fun_o': 'Fun',
    'amb_o': 'Ambitious'
}

# Replace the attributes names
partner_perception_general['Attributes'] = partner_perception_general['Attributes'].replace(attribute_mapping)
partner_perception_general


Attributes      Mean
0  Attractivity  6.190411
1     Sincerity  7.175256
2  Intelligence  7.369301
3           Fun  6.400599
4     Ambitious  6.778409

In [14]:
trace1 = go.Bar(
    x=own_perception_general["Attributes"],
    y=own_perception_general["Mean"],
    name = str("Own Evaluation")
)
trace2 = go.Bar(
    x=partner_perception_general["Attributes"],
    y=partner_perception_general["Mean"],
    name = str("Partner Evaluation")
)

fig = make_subplots()
fig.add_trace(trace1)
fig.add_trace(trace2)

fig.update_layout(title = "Comparison of evaluation : <b> Own evaluation Vs partner evaluation </b>",xaxis_title='Attributes', yaxis_title='Mean of Attributes (/10)', legend_title='Type of Evaluation',
                autosize=False,
                width=1000,
                height=512 )

fig.show()

There is a little difference from 1 point between the own evaluation and the partner evaluation.

Now we will see if there is a difference of that between the gender. 

With men:

In [9]:
list_of_attribut = ["attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1"]

# Filer the data where we have a male, gender = 1
df_male = df[df['gender'] == 1]

# calculate the mean for the attributes
own_perception_male = df_male[list_of_attribut].mean().reset_index(name="Mean")
own_perception_male = own_perception_male.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr3_1': 'Attractivity',
    'sinc3_1': 'Sincerity',
    'intel3_1': 'Intelligence',
    'fun3_1': 'Fun',
    'amb3_1': 'Ambitious'
}

# Replace the attributes names
own_perception_male['Attributes'] = own_perception_male['Attributes'].replace(attribute_mapping)
own_perception_male


Attributes      Mean
0  Attractivity  6.951636
1     Sincerity  8.133061
2  Intelligence  8.486526
3           Fun  7.517084
4     Ambitious  7.524783

In [10]:
list_of_attribut = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o"]

# Filer the data where we have a male, gender = 1
df_male = df[df['gender'] == 1]

# calculate the mean for the attributes
partner_perception_male = df_male[list_of_attribut].mean().reset_index(name="Mean")
partner_perception_male = partner_perception_male.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr_o': 'Attractivity',
    'sinc_o': 'Sincerity',
    'intel_o': 'Intelligence',
    'fun_o': 'Fun',
    'amb_o': 'Ambitious'
}

# Replace the attributes names
partner_perception_male['Attributes'] = partner_perception_male['Attributes'].replace(attribute_mapping)
partner_perception_male


Attributes      Mean
0  Attractivity  5.919422
1     Sincerity  7.099778
2  Intelligence  7.447362
3           Fun  6.280555
4     Ambitious  6.952773

In [24]:
trace1 = go.Bar(
    x=own_perception_male["Attributes"],
    y=own_perception_male["Mean"],
    name = str("Own Evaluation"),
    marker_color=px.colors.qualitative.Plotly[0] 
)
trace2 = go.Bar(
    x=partner_perception_male["Attributes"],
    y=partner_perception_male["Mean"],
    name = str("Partner Evaluation"),
    marker_color=px.colors.qualitative.Plotly[1] 
)

fig = make_subplots()
fig.add_trace(trace1)
fig.add_trace(trace2)

fig.update_layout(title = "Comparison of evaluation for men : <b> Own evaluation Vs partner evaluation </b>",xaxis_title='Attributes', yaxis_title='Mean of Attributes (/10)', legend_title='Type of Evaluation',
                autosize=False,
                width=1000,
                height=512 )

fig.show()

With women:

In [12]:
list_of_attribut = ["attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1"]

# Filer the data where we have a male, gender = 1
df_female = df[df['gender'] == 0]

# calculate the mean for the attributes
own_perception_female = df_female[list_of_attribut].mean().reset_index(name="Mean")
own_perception_female = own_perception_female.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr3_1': 'Attractivity',
    'sinc3_1': 'Sincerity',
    'intel3_1': 'Intelligence',
    'fun3_1': 'Fun',
    'amb3_1': 'Ambitious'
}

# Replace the attributes names
own_perception_female['Attributes'] = own_perception_female['Attributes'].replace(attribute_mapping)
own_perception_female

Attributes      Mean
0  Attractivity  7.219092
1     Sincerity  8.458343
2  Intelligence  8.320622
3           Fun  7.893612
4     Ambitious  7.632499

In [13]:
list_of_attribut = ["attr_o", "sinc_o", "intel_o", "fun_o", "amb_o"]

# Filer the data where we have a male, gender = 1
df_female = df[df['gender'] == 0]

# calculate the mean for the attributes
partner_perception_female = df_female[list_of_attribut].mean().reset_index(name="Mean")
partner_perception_female = partner_perception_female.rename(columns={'index': 'Attributes'})

# Dictionnary to change the name of attributes for mapping
attribute_mapping = {
    'attr_o': 'Attractivity',
    'sinc_o': 'Sincerity',
    'intel_o': 'Intelligence',
    'fun_o': 'Fun',
    'amb_o': 'Ambitious'
}

# Replace the attributes names
partner_perception_female['Attributes'] = partner_perception_female['Attributes'].replace(attribute_mapping)
partner_perception_female

Attributes      Mean
0  Attractivity  6.461401
1     Sincerity  7.251053
2  Intelligence  7.291202
3           Fun  6.520164
4     Ambitious  6.604591

In [25]:
trace1 = go.Bar(
    x=own_perception_female["Attributes"],
    y=own_perception_female["Mean"],
    name = str("Own Evaluation"),
    marker_color=px.colors.qualitative.Plotly[3] 
)
trace2 = go.Bar(
    x=partner_perception_female["Attributes"],
    y=partner_perception_female["Mean"],
    name = str("Partner Evaluation"),
    marker_color=px.colors.qualitative.Plotly[4] 
)

fig = make_subplots()
fig.add_trace(trace1)
fig.add_trace(trace2)

fig.update_layout(title = "Comparison of evaluation for women : <b> Own evaluation Vs partner evaluation </b>",xaxis_title='Attributes', yaxis_title='Mean of Attributes (/10)', legend_title='Type of Evaluation',
                autosize=False,
                width=1000,
                height=512 )

fig.show()

In [26]:
trace1 = go.Bar(
    x=own_perception_male["Attributes"],
    y=own_perception_male["Mean"],
    name = str("Own Evaluation for Men"),
    marker_color=px.colors.qualitative.Plotly[0] 
)
trace2 = go.Bar(
    x=partner_perception_male["Attributes"],
    y=partner_perception_male["Mean"],
    name = str("Partner Evaluation for Men"),
    marker_color=px.colors.qualitative.Plotly[1] 
)
trace3 = go.Bar(
    x=own_perception_female["Attributes"],
    y=own_perception_female["Mean"],
    name = str("Own Evaluation for Women"),
    marker_color=px.colors.qualitative.Plotly[3] 
)
trace4 = go.Bar(
    x=partner_perception_female["Attributes"],
    y=partner_perception_female["Mean"],
    name = str("Partner Evaluation for Women"),
    marker_color=px.colors.qualitative.Plotly[4] 
)

fig = make_subplots()
fig.add_trace(trace1)
fig.add_trace(trace3)
fig.add_trace(trace2)
fig.add_trace(trace4)

fig.update_layout(title = "Comparison of evaluation : <b> Own evaluation Vs partner evaluation </b> with <b> Men Vs Women <b>",xaxis_title='Attributes', yaxis_title='Mean of Attributes (/10)', legend_title='Type of Evaluation',
                autosize=False,
                width=1000,
                height=512 )

fig.show()

So, we see than the previous trend is similar in the two ways, for men and for women. We just see than women have a little higher own evaluation than men, and for the two genders it is oveestimated compared to the partner evaluation.

It may has a importance in the match choice or the second date continuation. If people considers themselves higher than the partner in front of them, they could choose to not continue with this partner that is not at their level whereas it is not really true.

# 6) 5th step, Importance of religion compare to other factors

In this section, we will try to see if there is a difference in importance between the religion and the 2 main attributes (attractiveness & fun). 

The importance of religion is rated on a scale 0 - 10 with: `imprelig`

For the importance of attractiveness and fun, we will use the same used before (scale 0 - 100): `attr7_2` & `fun7_2`

So we will need to compute the `imprelig` on the same scale than the othe parameter, on a %.

In [14]:
attractiveness = df["attr7_2"].mean()
print("The mean of attarctiveness importance score is:")
display(attractiveness)

fun = df["fun7_2"].mean()
print("The mean of fun importance score is:")
display(fun)

religion_imp = df["imprelig"].mean()
print("The mean of average religion importance score is:")
display(religion_imp/10 * 100) #calculation to put the value between 0 and 100 to have a percentage, like interests shared

data = {
    'category': ['Fun', 'Attractiveness', 'Religion importance'],
    'mean': [fun, attractiveness, religion_imp/10 *100]
}

importance_attributes_vs_religion = pd.DataFrame(data)

display(importance_attributes_vs_religion)

The mean of attarctiveness importance score is:


32.819556451612904

The mean of fun importance score is:


18.86844758064516

The mean of average religion importance score is:


36.51644776479094

category       mean
0                  Fun  18.868448
1       Attractiveness  32.819556
2  Religion importance  36.516448

In [23]:
fig = px.bar(importance_attributes_vs_religion, x = "category", y = "mean", color = 'category', color_discrete_sequence=px.colors.qualitative.Pastel[1:4])
fig.update_layout(
    title = "Comparison of importance : <b> High scored attributes Vs Religion importance </b>",
    xaxis_title = "Category for comparison",
    yaxis_title = "Mean score of importance (%)",
    autosize=False,
    width=1000,
    height=512,
    showlegend=False,
    xaxis={'categoryorder':'total ascending'}
)

fig.show()

So we see here than the religion seems a more important criteria compared to some personal attributes, here the higher ones, the fun of a person and the attractiveness of this person. 

# 7) 6th step, Type of studies successful in matchs

Now, let's try to see if the domain of studies have an importance for matching.

The field of study is noted this way:  
`field`:     field of study   
`Law`    
`Math`  
`Social Science, Psychologist`   
`Medical Science, Pharmaceuticals, and Bio Tech`   
`Engineering`    
`English/Creative Writing/ Journalism`   
`History/Religion/Philosophy`   
`Business/Econ/Finance`   
`Education, Academia`   
`Biological Sciences/Chemistry/Physics`  
`Social Work`   
`Undergrad/undecided`   
`Political Science/International Affairs`   
`Film`  
`Fine Arts/Arts Administration`  
`Languages`  
`Architecture`  
`Other`  


And the match factor is just `match` = `1` for "Yes" or `0` for "No"


In [25]:
total_per_field = df['field'].value_counts().reset_index(name="Total Sum per field of study")

df_match = df[df["match"] == 1]
match_per_field = df_match['field'].value_counts().reset_index(name="Total Sum per field of study with match")

match_field_of_study = match_per_field.merge(total_per_field, on='field')

match_field_of_study["Proportion of match per field of study"] = match_field_of_study["Total Sum per field of study with match"] / match_field_of_study["Total Sum per field of study"] * 100

display(match_field_of_study)

field  \
0                                  Business   
1                                       Law   
2                                       MBA   
3                               Social Work   
4                     International Affairs   
..                                      ...   
216                           Public Policy   
217                            Business/Law   
218                       Finance&Economics   
219  International Finance, Economic Policy   
220                  Masters of Social Work   

     Total Sum per field of study with match  Total Sum per field of study  \
0                                        111                           521   
1                                         98                           462   
2                                         69                           468   
3                                         45                           378   
4                                         36                           252   
..                                       ...                           ...   
216                                        1                            10   
217                                        1                            10   
218                                        1                            19   
219                                        1                            10   
220                                        1                            18   

     Proportion of match per field of study  
0                                 21.305182  
1                                 21.212121  
2                                 14.743590  
3                                 11.904762  
4                                 14.285714  
..                                      ...  
216                               10.000000  
217                               10.000000  
218                                5.263158  
219                               10.000000  
220                                5.555556  

[221 rows x 4 columns]

In [5]:
fig = px.bar(match_field_of_study.loc[:10], x = "field", y = "Total Sum per field of study with match", color_discrete_sequence=px.colors.qualitative.Pastel[5:6])
fig.update_layout(
    title = "Sum of match in function of the field of study",
    xaxis_title = "Field of study",
    yaxis_title = "Count of match",
    autosize=False,
    width=1000,
    height=512,
    showlegend=False,
    xaxis={'categoryorder':'total ascending'}
)

fig.show()

Here in the data, we see there is 2 Law with "Law" and "law" ; and two Physics with "Physics" and "physics". There is no explication in the doc of data on thatn so I think it is the same fields entered with a mistake of forgotten capital letter. So we will add the calculated values from "law" line into "Law", idem with "Physics". 

With that, we will observe more respectful results.

In [36]:
total_per_field = df['field'].value_counts().reset_index(name="Total Sum per field of study")

df_match = df[df["match"] == 1]
match_per_field = df_match['field'].value_counts().reset_index(name="Total Sum per field of study with match")

match_field_of_study = match_per_field.merge(total_per_field, on='field')

#add together the lines with fields with Law & law; Physics & physics
match_field_of_study.loc[match_field_of_study['field'] == 'Law', 'Total Sum per field of study'] += match_field_of_study.loc[match_field_of_study['field'].isin(['law']), 'Total Sum per field of study'].sum()
match_field_of_study.loc[match_field_of_study['field'] == 'Physics', 'Total Sum per field of study'] += match_field_of_study.loc[match_field_of_study['field'].isin(['physics']), 'Total Sum per field of study'].sum()
match_field_of_study.loc[match_field_of_study['field'] == 'Law', 'Total Sum per field of study with match'] += match_field_of_study.loc[match_field_of_study['field'].isin(['law']), 'Total Sum per field of study with match'].sum()
match_field_of_study.loc[match_field_of_study['field'] == 'Physics', 'Total Sum per field of study with match'] += match_field_of_study.loc[match_field_of_study['field'].isin(['physics']), 'Total Sum per field of study with match'].sum()
#Delete the lines with field = law & physics that are useless now
match_field_of_study = match_field_of_study.drop(match_field_of_study[match_field_of_study['field'] == 'law'].index)
match_field_of_study = match_field_of_study.drop(match_field_of_study[match_field_of_study['field'] == 'physics'].index)

match_field_of_study["Proportion of match per field of study"] = match_field_of_study["Total Sum per field of study with match"] / match_field_of_study["Total Sum per field of study"] * 100

display(match_field_of_study)

field  \
0                                  Business   
1                                       Law   
2                                       MBA   
3                               Social Work   
4                     International Affairs   
..                                      ...   
216                           Public Policy   
217                            Business/Law   
218                       Finance&Economics   
219  International Finance, Economic Policy   
220                  Masters of Social Work   

     Total Sum per field of study with match  Total Sum per field of study  \
0                                        111                           521   
1                                        126                           585   
2                                         69                           468   
3                                         45                           378   
4                                         36                           252   
..                                       ...                           ...   
216                                        1                            10   
217                                        1                            10   
218                                        1                            19   
219                                        1                            10   
220                                        1                            18   

     Proportion of match per field of study  
0                                 21.305182  
1                                 21.538462  
2                                 14.743590  
3                                 11.904762  
4                                 14.285714  
..                                      ...  
216                               10.000000  
217                               10.000000  
218                                5.263158  
219                               10.000000  
220                                5.555556  

[219 rows x 4 columns]

In [37]:
fig = px.bar(match_field_of_study.loc[:10], x = "field", y = "Total Sum per field of study with match", color_discrete_sequence=px.colors.qualitative.Pastel[5:6])
fig.update_layout(
    title = "Sum of match in function of the field of study",
    xaxis_title = "Field of study",
    yaxis_title = "Count of match",
    autosize=False,
    width=1000,
    height=512,
    showlegend=False,
    xaxis={'categoryorder':'total ascending'}
)

fig.show()

In [38]:
fig = px.bar(match_field_of_study.loc[:10], x = "field", y = "Proportion of match per field of study", color_discrete_sequence=px.colors.qualitative.Pastel[3:4])
fig.update_layout(
    title = "Proportion of match in function of the field of study",
    xaxis_title = "Field of study",
    yaxis_title = "Proportion of match (%)",
    autosize=False,
    width=1000,
    height=512,
    showlegend=False,
    xaxis={'categoryorder':'total ascending'}
)

fig.show()

With these 2 plots, we see that there is more Law students with a match but they are also a lot. So the 1rst plot doesn't really give a potential usefull information. However, the 2nd plot with the proportion of match in function of the field of study show than medicine student have a higher chance to match. 

It can be explain by their field of study with these different characteristics as hypotheses:

- They are more empathic due to their field of study.
- They have more social skills.
- There is an higher chance that they will earn a good living.


# 8) Final step, some conclusions

To conclude and help the app tinder to make the good choice to propose people for a match that will continu on a second date, we observed the following points.

🌟 For both of gender, the more important attribute is the attractiveness, following by the fun of the partner. But everyone tends to have an higher own evaluation than what the partner evaluates. So without external help (like an AI object that can evaluates the attractiveness based on a photo and notes it for help the algorithm to propose people aroud the same note) it will be complicated. Maybe during the inscription, the app could ask more external informations to know what the person consider as attractive. One more thing, the attarctiveness importance is underestimated before to see people in real.

🌟 The shared racial background is pretty high. It is more important to have the same racial background than the same hobbies for the majority. So the algorithm should take this in count to propose the profile. And it should be indiquated during the profil creation.

🌟 In the same way, it is the same thing for religion. People are more able to have a match or accept multiple date if they have the same religion, which is link to the life philosophy of the person. So it should be considered in the profil creation and with the algorithm. It has be shown by the comparison with the 2 higher attributes condidered as important. The same religion background is higher in importance.

🌟 To finish the field of study/work count. It is also due to than it is linked to the person character, philosophy, mind, emotional intelligence... That's why medicine students have more match in proportion of their total number. So maybe the app should more condider theses characteristics in people, and engage people in this way and not just a human consomption. It is more durable and effecdtive.